types/react/index.d.ts

```diff
@@ -137,10 +137,10 @@ declare namespace React {
         key?: Key;
     }
     interface RefAttributes<T> extends Attributes {
-        ref?: Ref<T>;
+        ref?: Ref<T> | MutableRefObject<T | undefined>;
```

Comment:
I initially tried to fix that by changing LegacyRef/Ref, but that broke ElementRef. That's the best I could get without breaking everything. Will gladly accept inputs if there's a cleaner way to solve that :)